spaceWeatherApi/Controllers/HomeController.cs

```diff
@@ -1,197 +0,0 @@
-                        if (cmeRegion == null || solarRegion == null) return false; 
-
-                        return cmeRegion.EndsWith(solarRegion) && c.StartTime >= thirtyDaysAgo;
-                    }),
-                    RecentCMECount = allCMEEvents.Count(c => c.ActiveRegionNum == g.Key && c.StartTime >= sevenDaysAgo)
-                })
-                   .OrderByDescending(r => r.TotalSunspots + r.SignificantFlares.Count * 10 + r.CMECount * 5).ToList();
-
-
-
-            //CME data can contain a lot of nulls, this gives its own entry in the report 
-            var nullRegionCMEs = allCMEEvents.Count(c => c.ActiveRegionNum == null && c.StartTime >= thirtyDaysAgo);
-            if (nullRegionCMEs > 0)
-            {
-                report.Add(new
-                {
-                    Region = (int?) -1,  
-                    TotalSunspots = 0d,
-                    RecentSunspots = 0d,
-                    SignificantFlares = new List<FlareEvent>(),
-                    RecentSignificantFlares = new List<FlareEvent>(),
-                    CMECount = nullRegionCMEs,
-                    RecentCMECount = allCMEEvents.Count(c => c.ActiveRegionNum == null && c.StartTime >= sevenDaysAgo)
-                });
-            }
-
-
-            // Create a text report
-            var textReport = new StringBuilder();
-            // Add solar flux data
-            var fluxData = await _apiClient.GetTodayFluxNum();
-            if (fluxData != null && !string.IsNullOrEmpty(fluxData.Flux))
-            {
-                //sfu = 10-22W m-2 Hz-1)
-                textReport.AppendLine($"Solar Flux: {fluxData.Flux} sfu");
-            }
-            else
-            {
-                textReport.AppendLine("Solar Flux: N/A");
-            }
-
-            textReport.AppendLine();
-            textReport.AppendLine("** 30 days worth of data, ordered by last 7 days activity **");
-            textReport.AppendLine("| Region | Total SSN | M/X Flares | Strongest Flare | CME | Note |");
-            textReport.AppendLine("|--------|-----------|------------|-----------------|-----|------|");
-
-            foreach (var region in report)
-            {
-                string? regionDisplay = region.Region == -1 ? "NORE" : region.Region.ToString();
-                var strongestFlare = region.SignificantFlares.OrderByDescending(f => f.ClassType).FirstOrDefault();
-                var activityTrend = DetermineActivityTrend(region.TotalSunspots, region.SignificantFlares.Count, region.CMECount);
-
-                textReport.AppendLine($"| {regionDisplay,6} | {region.TotalSunspots,9:F0} | {region.SignificantFlares.Count,10} | {strongestFlare?.ClassType ?? "N/A",15} | {region.CMECount,3} | {activityTrend,-20} |");
-            }
-            return Ok(textReport.ToString());
-        }
-
-
-        private static string DetermineActivityTrend(double sunspots, int flares, int cmes)
```

Comment:
this works. but you could research pattern matching for more modern tools and syntax